### PR TITLE
optimize enumeration of network interfaces on windows

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: src/enum_net.cpp [hunk 1]
```diff
@@ -552,54 +552,43 @@ namespace libtorrent
...
 
-				adapter_addresses = (IP_ADAPTER_ADDRESSES*)malloc(out_buf_size);
-				if (!adapter_addresses)
-				{
-					FreeLibrary(iphlp);
-					ec = asio::error::no_memory;
-					return std::vector<ip_interface>();
-				}
+			ULONG buf_size = 10000;
+			std::vector<char> buffer(buf_size);
```
Comment (Robo210) [inline src/enum_net.cpp]: I believe it is still possible for a larger buffer to be required, so you may want to keep the ERROR_BUFFER_OVERFLOW check and resize if needed. In the common case though this should avoid the need for actually making this second call.

Comment (arvidn) [inline src/enum_net.cpp]: updated patch

Comment (Robo210) [inline src/enum_net.cpp]: The fix seems good to me.

File: src/enum_net.cpp [hunk 2]
```diff
@@ -552,54 +552,55 @@ namespace libtorrent
...
+			}
 
-				adapter_addresses = (IP_ADAPTER_ADDRESSES*)malloc(out_buf_size);
-				if (!adapter_addresses)
-				{
-					FreeLibrary(iphlp);
-					ec = asio::error::no_memory;
-					return std::vector<ip_interface>();
-				}
+			ULONG buf_size = 10000;
```
Comment (Chocobo1) [inline src/enum_net.cpp]: I guess you already know this, quoting MSDN:
```
The recommended method of calling the GetAdaptersAddresses function is to pre-allocate a 15KB working buffer pointed to by the AdapterAddresses parameter. 
```
Not sure should you really follow that...

Comment (arvidn) [inline src/enum_net.cpp]: well, 15 felt a bit excessive, so I went with 10.

**Pull request merged**